src/units/abilities.cpp

```diff
@@ -1168,6 +1168,48 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 	}
 }
 
+//Functions used for change damage_type list with damage
+static std::vector<std::string> vector_type(unit_ability_list& abil_list, const std::string& type)
+{
+	std::vector<std::string> type_list;
+	for(auto& i : abil_list) {
+		if(!(*i.ability_cfg)[type].str().empty()){
+			type_list.push_back((*i.ability_cfg)[type].str());
+		}
+	}
+	if(type_list.size() >= 2){
+		std::sort(type_list.begin(), type_list.end());
+		if(type_list.size() >= 3){
+			std::unordered_map<std::string, unsigned int> type_count;
+			for( const std::string& character : type_list ){
+				type_count[character]++;
+			}
+			std::sort( std::begin( type_list ) , std::end( type_list ) , [&]( const std::string& rhs , const std::string& lhs ){
+				return type_count[lhs] < type_count[rhs];
+			});
+		}
+	}
```

Comment:
This should be:
```
	if(type_list.size() == 2){
		std::sort(type_list.begin(), type_list.end());
	} else if(type_list.size() >= 3){
		std::unordered_map<std::string, unsigned int> type_count;
		for( const std::string& character : type_list ){
			type_count[character]++;
		}
		std::sort( std::begin( type_list ) , std::end( type_list ) , [&]( const std::string& rhs , const std::string& lhs ){
			return type_count[lhs] < type_count[rhs];
		});
	}
```
otherwise you sort all lists with 3 or more elements twice.

Reply:
done.